src/sources/kubernetes/mod.rs

```diff
@@ -105,48 +119,161 @@ impl TimeFilter {
+            // Will include logs of Kubernetes < v1.14.
+            file_config.include.push(
+                (LOG_DIRECTORY.to_owned()
+                    + format!("{}/{}*/*.log", object_uid, container_name).as_str())
+                .into(),
+            );
+
+            for namespace in flatten_include(&config.include_namespaces, &any) {
+                // Will include logs of Kubernetes >= v1.14.
+                file_config.include.push(
+                    (LOG_DIRECTORY.to_owned()
+                        + format!("{}_*_{}/{}*/*.log", namespace, object_uid, container_name)
+                            .as_str())
+                    .into(),
+                );
+            }
+        }
+    }
+
+    Ok(())
+}
+
+// Parses partial UIDs and transforms them to two UID forms with glob [0-9A-Fa-f] filling
+// them up to 32 hexadecimal numbers.
+fn prepare_pod_uids(include: &Option<Vec<String>>) -> Result<Vec<String>, BuildError> {
+    let mut include_pod_uids = Vec::new();
+    for pod_uid in flatten_include(include, &[]) {
+        let mut pod_uid = pod_uid.clone();
+
+        if !pod_uid.chars().all(|c| {
+            c.is_numeric() || ('A'..='F').contains(&c) || ('a'..='f').contains(&c) || c == '-'
+        }) {
+            error!(message = "Configuration 'include_pod_uids' contains not UID", uid = ?pod_uid);
+            return Err(BuildError::IllegalCharacterInUid {
+                uid: pod_uid.clone(),
+            });
+        }
+
+        pod_uid.retain(|c| c != '-');
+
+        if pod_uid.chars().count() > 32 {
+            return Err(BuildError::UidToLarge {
+                uid: pod_uid.clone(),
+            });
+        }
+
+        include_pod_uids.append(&mut to_uid_forms(pod_uid.as_str()));
+    }
+
+    Ok(include_pod_uids)
+}
+
+fn file_source_exclude(config: &KubernetesConfig, file_config: &mut FileConfig) {
+    let no_include = flatten_include(&config.include_container_names, &[]).is_empty()
+        && flatten_include(&config.include_namespaces, &[]).is_empty()
+        && flatten_include(&config.include_pod_uids, &[]).is_empty();
+
+    // Default excludes
+    if no_include {
+        // Since there is no user intention in including specific namespace/pod/container,
+        // exclude kuberenetes and vector logs.
+
+        // This is correct, but on best effort basis filtering out of logs from kuberentes system components.
+        // More specificly, it will work for all Kubernetes 1.14 and higher, and for some bellow that.
+        file_config
+            .exclude
+            .push((LOG_DIRECTORY.to_owned() + r"kube-system_*").into());
+
+        // NOTE: for now exclude images with name vector, it's a rough solution, but necessary for now
+        file_config
+            .exclude
+            .push((LOG_DIRECTORY.to_owned() + r"*/vector*").into());
+    }
+}
+
+fn flatten_include<'a>(o: &'a Option<Vec<String>>, default: &'a [String]) -> &'a [String] {
```

Comment:
not sure why we need  a function for this and I also don't think you need to convert it to a slice, you can just check the len of the vec?

Reply:
It might make sense to separate where we flatten things and where we check for include/excludes, the logic is a bit overlapping here.

Reply:
It's a helper function, exists just to eliminate code duplication. Although, the name is really confusing. It's basically a `not_empty_or_else` function. I'll change the name, and simplify it.